recipes/libelfin/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+import os
+
+
+class LibelfinConan(ConanFile):
+    name = "libelfin"
+    description = "C++11 library for reading ELF binaries and DWARFv4 debug information"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aclements/libelfin"
+    license = "MIT"
+    topics = ("conan", "elf", "dwarf", "libelfin")
+
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    exports_sources = "patches/*"
+
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        with tools.chdir(self._source_subfolder):
+            autotools = AutoToolsBuildEnvironment(self)
+            autotools.make()
+            autotools.install(args=["PREFIX=" + os.path.join(self.source_folder, "package")])
```

Comment:
Please check other recipes using `AUtotoolsBuildEnvironment`:
- cache the autotools object with a `_configure_autotools` method
- pass some arguments to the `configure` script (`--enable-shared=yes/no`, `--enable-static=yes/no`)
- move the `install` to `package`